### PR TITLE
Add Visualizers

### PR DESCRIPTION
**Pull request by ueda0319**
## Issue URL

#46 #48 

## Change overview

Add visualizer of each field for pretrained NeDDF.
Mesh generator with MarchingCube.
Mesh viewer

## How to test

`poetry run python neddf/scripts/fields_visualizer.py pretrained/lego/`

## Note for reviewers

Visualization demo video is provided as follow(It can also be on README)

https://user-images.githubusercontent.com/26667016/184655073-1ca5f55c-0170-4c61-96aa-802aa9ccfc88.mp4

### CODE REVIEW

File: neddf/scripts/fields_visualizer.py [hunk 1]
```diff
@@ -0,0 +1,596 @@
...
+        transform[0, 2] = -1
+        transform[1, 0] = -1
+        transform[2, 1] = 1
+        transform[3, 3] = 1
+        self.meshed_field.transform(transform)
+
+        # mesh_file_path: Path = mesh_dir / "mesh_{}_threshold{}.dae".format(
+        #     cube_resolution, threshold
+        # )
+        # mcubes.export_mesh(vertices, triangles, mesh_file_path.as_posix, "mcube")
```
Comment (gatheluck) [inline neddf/scripts/fields_visualizer.py:553]: @ueda0319 I think these lines should be un-commented for save exported mesh file.

**Comment by gatheluck** [comment]
@ueda0319 
Sorry, could you teach me what should be expected when; 
(i) we press "Generate mesh model" button?
(ii) we press "Refresh render model" button?

About (i) case, I could not find difference after press the button. After that I pressed "Refresh render model" button multiple times, then long gray lines (axis?) appeared. I'm not sure but it doesn't intended behavior, right?
Screenshot from 2022-08-16 22-07-40

**Comment by ueda0319** [comment]
@gatheluck Sorry for the confusing layout.
(i) The "Generate mesh model" button regenerates the mesh according to the marching cube resolution and threshold.
(ii) The "Refresh render model" button reloads the mesh model when the view is not updated properly, so there is basically no need to press this button. It seems to be working without this button at present, so I remove it.
I update the layout to use collapses as following
Screenshot from 2022-08-17 10-22-20

File: neddf/scripts/fields_visualizer.py (hunk 1 above)
Comment (ueda0319) [reply neddf/scripts/fields_visualizer.py:553]: Thank you! I updated it on PR #51 , please confirm.

**Review by gatheluck — APPROVED** [review]
Thank you for your update, LGTM! Sometime I still see some very long gray lines. So I guess there still might be small bug. However it's rare case and I can't find the static way to reproduce it. So I think it's OK to release at this point. And we can wait bug report.

**Pull request merged**